server/test/list_objects.go

```diff
@@ -161,6 +161,17 @@ func runListObjectsTests(t *testing.T, ctx context.Context, testCases []listObje
 	for _, test := range testCases {
 		t.Run(test.name+"/streaming", func(t *testing.T) {
 			server := NewMockStreamServer(len(test.expectedResult))
+
+			done := make(chan struct{})
+			var streamedObjectIds []string
+			go func() {
+				for x := range server.channel {
+					streamedObjectIds = append(streamedObjectIds, x)
+				}
+
+				done <- struct{}{}
```

Comment:
@miparnisari this is the trick. `ExecuteStreamed` returns when all child routines are finished, and then we close the channel with `close(server.channel)`, but it's possible we may have not read everything emitted on the channel.

Blocking on the `done` channel below and emitting it only after we've fully consumed the channel is the best way to ensure we consume everything before checking our results.